python/tvm/relay/frontend/tensorflow_ops.py

```diff
@@ -1162,6 +1163,9 @@ def _impl(inputs, attr, params, mod):
         adj_x = attr["adj_x"]
         adj_y = attr["adj_y"]
         input_x = _op.transpose(input_x, axes=[0, 2, 1]) if adj_x else input_x
+        shape_y = _infer_shape(input_y, mod)
+        if len(shape_y) < 3:
+            input_y = _op.reshape(input_y, (1, orig_shape_y[-2], orig_shape_y[-1]))
```

Comment:
Is this required for static shape? It seems to me that you'll get two reshapes, although it should be simplified later by the SimplifyExpr pass.
```
%1 = reshape(%y, (1, p, q));
%2 = reshape(%1, (1, p, q));
```

Reply:
Thanks @comaniac . It is important for cases where `ndim=len(shape_x)` is <=3. I have test cases that fail without this line. Like `_test_batch_matmul((1, 8, 64), (64, 1), "float32", False, False)`.  However the case that you mentioned can also happen for certain input configurations. I refactored the logic to avoid that.